cci_cameroon/analysis/text_similarity/similarity_search_sample.py

```diff
@@ -0,0 +1,570 @@
+#     name: python3
+# ---
+
+# %%
+import pandas as pd
+import os
+import scipy
+import tensorflow as tf
+import faiss
+from sentence_transformers import SentenceTransformer, util
+import requests
+from io import StringIO
+import time
+import logging
+import re
+
+# %matplotlib inline
+import torch
+import regex
+import omegaconf
+import numpy as np
+import cci_cameroon
+from cci_cameroon.getters.external_data import (
+    fetch_data,
+    similarity,
+    compute_embedding_cosign_score,
+)
+
+# %%
+# !pip list
+
+# %% [markdown]
+# # Semantic Textual Similarity matching
+# This notebook contains an implementation of sample transformer models for both English and French text. Text Embeddings are generated using different pre-trained models and their performance on matching sample strings is evaluated. The cosine similarity score or the Euclidean distance between the created vectors is used for the evaluation.
+
+# %%
+# install sentence tranformers library using command below
+# conda install -c conda-forge sentence-transformers
+# pip install hydra-core omegaconf
+# #!pip install bitarray
+# #!pip install --no-cache-dir sentencepiece
+base_dir = cci_cameroon.PROJECT_DIR
+
+
+# %%
+urls = []
+
+# %%
+# read in sample text dataset from online.
+with open(f"{base_dir}/inputs/data/english_url.txt") as f:
+    urls = [u.strip("\n") for u in f]
+sentences = fetch_data(urls)
+
+# %%
+sentences[:10]
+
+# %% [markdown]
+# ## Removing word duplicates and NAN from the collected data
+
+# %%
+# preprocessing the sentences fetched by removing duplicates and NaN
+sentences = [word for word in list(set(sentences)) if type(word) is str]
+
+# %% [markdown]
+# ## initialize sentence transformer model - using BERT. Sentence embedding takes time
+# The bert-base-nli-mean-tokens pre-trained model initialized here and used to create sentence embeddings.
+
+# %%
+# initialize sentence transformer model - using BERT. Sentence embedding takes time
+model = SentenceTransformer(
+    "bert-base-nli-mean-tokens"
+)  # loading simple pre-trained transformer model for learning purposes
+# create sentence embeddings
+sentence_embeddings = model.encode(sentences)
+sentence_embeddings.shape
+
+# %%
+dimension = sentence_embeddings.shape[1]
+
+# %% [markdown]
+# ## Faiss flat index
+# We create a faiss flat index to search the vector embeddings for a match with a new string.
+# The Euclidean distance is used to measure similarity between strings. The shorter the distance,
+# the greater the similarity.
+
+# %%
+# initialize the index with the dimension of the dataset
+index = faiss.IndexFlatL2(dimension)
+
+# %%
+sentence_embeddings.shape
+
+# %%
+index.is_trained
+
+# %%
+# we load the embeddings to the index
+index.add(sentence_embeddings)
+index.ntotal
+
+# %%
+sentences[:15]
+
+
+# %% [markdown]
+# ## readInput
+# a function to read in sample string used in the match search.
+
+# %%
+# This function allows a user to input a search string for testing
+def readInput():
```

Comment:
```suggestion
def read_input():
```

PEP8  preference is to use lower snake case for function names https://www.python.org/dev/peps/pep-0008/#function-and-variable-names